src/opm/input/eclipse/Schedule/UDQ/UDQConfig.cpp

```diff
@@ -480,25 +469,30 @@ namespace Opm {
                                 UDQState&         udq_state,
                                 UDQContext&       context) const
     {
-        for (const auto& def : this->definitions(UDQVarType::WELL_VAR)) {
-            if (udq_state.define(def.keyword(), def.status())) {
-                auto ws = def.eval(context);
-                context.update_define(report_step, def.keyword(), ws);
-            }
-        }
+        auto var_type_bit = [](const UDQVarType var_type)
+        {
+            return 1ul << static_cast<std::size_t>(var_type);
+        };
+
+        auto select_var_type = std::size_t{0};
+        select_var_type |= var_type_bit(UDQVarType::WELL_VAR);
+        select_var_type |= var_type_bit(UDQVarType::GROUP_VAR);
+        select_var_type |= var_type_bit(UDQVarType::FIELD_VAR);
 
-        for (const auto& def : this->definitions(UDQVarType::GROUP_VAR)) {
-            if (udq_state.define(def.keyword(), def.status())) {
-                auto ws = def.eval(context);
-                context.update_define(report_step, def.keyword(), ws);
+        for (const auto& [keyword, index] : this->input_index) {
+            if (index.action != UDQAction::DEFINE) {
+                continue;
             }
-        }
 
-        for (const auto& def : this->definitions(UDQVarType::FIELD_VAR)) {
-            if (udq_state.define(def.keyword(), def.status())) {
-                auto field_udq = def.eval(context);
-                context.update_define(report_step, def.keyword(), field_udq);
+            auto def_pos = this->m_definitions.find(keyword);
+            if ((def_pos == this->m_definitions.end()) || // No such def
```

Comment:
Is this semantic change intentional?
Previously an exception would have been thrown if there  is no entry in definitions. Now we silently continue

Reply:
> Is this semantic change intentional?

Thanks for noticing.  That wasn't an intended part of the change.  I've reinstated the exception, but this time with a (hopefully) more informative diagnostic message than before.  Previously we would have gotten something like
```
Internal error: _Map_base::at
```
and we already have enough of those, e.g., OPM/opm-simulators#3806.